packages/legend-graph/src/helpers/DomainHelper.ts

```diff
@@ -49,3 +59,270 @@ export const getMilestoneTemporalStereotype = (
+              },
+            },
+            {
+              _type: 'var',
+              class: 'Date',
+              name: 'end',
+              multiplicity: {
+                lowerBound: 1,
+                upperBound: 1,
+              },
+            },
+          ];
+          milestonedAllVersionsInRange.body = [
+            {
+              _type: 'property',
+              parameters: [
+                {
+                  _type: 'var',
+                  name: 'this',
+                },
+              ],
+              property: `${property.name}`,
+            },
+          ];
+          propertyOwner._generatedMilestonedProperties.push(dateProperty);
+          propertyOwner._generatedMilestonedProperties.push(
+            milestonedAllVersions,
+          );
+          propertyOwner._generatedMilestonedProperties.push(
+            milestonedAllVersionsInRange,
+          );
+          break;
+        }
+        case MILESTONING_STEROTYPES.BITEMPORAL: {
+          const dateProperty = new DerivedProperty(
+            `${property.name}`,
+            graph.getTypicalMultiplicity(TYPICAL_MULTIPLICITY_TYPE.ONE),
+            GenericTypeExplicitReference.create(
+              new GenericType(
+                graph.getClass(property.genericType.value.rawType.path),
+              ),
+            ),
+            property.owner,
+          );
+          dateProperty.parameters = [
+            {
+              _type: 'var',
+              class: 'Date',
+              name: DEFAULT_MILESTONING_PARAMETERS.PROCESSING_DATE,
+              multiplicity: {
+                lowerBound: 1,
+                upperBound: 1,
+              },
+            },
+            {
+              _type: 'var',
+              class: 'Date',
+              name: DEFAULT_MILESTONING_PARAMETERS.BUSINESS_DATE,
+              multiplicity: {
+                lowerBound: 1,
+                upperBound: 1,
+              },
+            },
+          ];
+          dateProperty.body = [
+            {
+              _type: 'property',
+              parameters: [
+                {
+                  _type: 'var',
+                  name: 'this',
+                },
+              ],
+              property: `${property.name}`,
+            },
+          ];
+          const milestonedAllVersions = new Property(
+            `${property.name}AllVersions`,
+            graph.getTypicalMultiplicity(TYPICAL_MULTIPLICITY_TYPE.ONEMANY),
+            GenericTypeExplicitReference.create(
+              new GenericType(
+                graph.getClass(property.genericType.value.rawType.path),
+              ),
+            ),
+            property.owner,
+          );
+          propertyOwner._generatedMilestonedProperties.push(dateProperty);
+          propertyOwner._generatedMilestonedProperties.push(
+            milestonedAllVersions,
+          );
+          break;
+        }
+        default:
+          break;
+      }
+    }
+  });
+};
```

Comment:
Could we add a test case in `GraphBuildSuccess.test` testing the graph. Specifically testing the milestoned properties are generated.